Lib/fontbakery/specifications/googlefonts.py

```diff
@@ -998,3 +1004,155 @@ def check_description_strings_do_not_exceed_100_chars(fb, ttFont):
+              " %d instead.").format(width_max,
+                                     ttFont['hhea'].advanceWidthMax))
+  if is_monospaced:
+    if ttFont['post'].isFixedPitch != IS_FIXED_WIDTH_MONOSPACED:
+      failed = True
+      fb.error(("On monospaced fonts, the value of"
+                "post.isFixedPitch must be set to %d"
+                " (fixed width monospaced),"
+                " but got %d instead.").format(IS_FIXED_WIDTH_MONOSPACED,
+                                               ttFont['post'].isFixedPitch))
+
+    if ttFont['OS/2'].panose.bProportion != PANOSE_PROPORTION_MONOSPACED:
+      failed = True
+      fb.error(("On monospaced fonts, the value of"
+                "OS/2.panose.bProportion must be set to %d"
+                " (proportion: monospaced), but got"
+                " %d instead.").format(PANOSE_PROPORTION_MONOSPACED,
+                                       ttFont['OS/2'].panose.bProportion))
+
+    num_glyphs = len(ttFont['glyf'].glyphs)
+    outliers = num_glyphs - stats['most_common']['occurrences']
+    if outliers > 0:
+      # If any glyphs are outliers, note them
+      unusually_spaced_glyphs = \
+       [g for g in glyphs
+        if font['hmtx'].metrics[g][0] != stats['most_common']['width']]
+      outliers_ratio = float(stats['most_common']['occurrences'])/num_glyphs
+
+      for glyphname in ['.notdef', '.null', 'NULL']:
+        if glyphname in unusually_spaced_glyphs:
+          unusually_spaced_glyphs.remove(glyphname)
+
+      failed = True
+      fb.warning(("Font is monospaced but {} glyphs"
+                  " ({}%) have a different width."
+                  " You should check the widths of: {}").format(
+                    outliers,
+                    100 - (100.0 * outliers_ratio),
+                    unusually_spaced_glyphs))
+    if not failed:
+      fb.ok("Font is monospaced and all related metadata look good.")
+  else:
+    # it is a non-monospaced font, so lets make sure
+    # that all monospace-related metadata is properly unset.
+
+    if ttFont['post'].isFixedPitch != IS_FIXED_WIDTH_NOT_MONOSPACED:
+      failed = True
+      fb.error(("On non-monospaced fonts, the"
+                " post.isFixedPitch value must be set to %d"
+                " (fixed width not monospaced), but got"
+                " %d instead.").format(IS_FIXED_WIDTH_NOT_MONOSPACED,
+                                       ttFont['post'].isFixedPitch))
+
+    if ttFont['OS/2'].panose.bProportion == PANOSE_PROPORTION_MONOSPACED:
+      failed = True
+      fb.error(("On non-monospaced fonts, the"
+                " OS/2.panose.bProportion value must be set to %d"
+                " (proportion: any), but got"
+                " %d (proportion: monospaced)"
+                " instead.").format(PANOSE_PROPORTION_ANY,
+                                    PANOSE_PROPORTION_MONOSPACED))
+    if not failed:
+      fb.ok("Font is not monospaced and all related metadata look good.")
```

Comment:
The test as it is looks good to me. The one thing I wonder most about is how many false positives the 80% rule produces.